apps/backend/src/controllers/agri/initiate/confirm.ts

```diff
@@ -0,0 +1,152 @@
+import { NextFunction, Request, Response } from "express";
+import { v4 as uuidv4 } from "uuid";
+import {
+	MOCKSERVER_ID,
+	send_response,
+	send_nack,
+	redisFetchToServer,
+	AGRI_BAP_MOCKSERVER_URL,
+	redis,
+	logger,
+} from "../../../lib/utils";
+import {
+	ACTTION_KEY,
+	ON_ACTION_KEY,
+} from "../../../lib/utils/actionOnActionKeys";
+import { ERROR_MESSAGES } from "../../../lib/utils/responseMessages";
+import { ORDER_STATUS, PAYMENT_STATUS } from "../../../lib/utils/apiConstants";
+
+export const initiateConfirmController = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+) => {
+	try {
+		const { scenario, transactionId } = req.body;
+		const on_search = await redisFetchToServer(
+			ON_ACTION_KEY.ON_SEARCH,
+			transactionId
+		);
+		const providersItems =
+			on_search?.message?.catalog["bpp/providers"][0]?.items;
+		const on_init = await redisFetchToServer(
+			ON_ACTION_KEY.ON_INIT,
+			transactionId
+		);
+		if (!on_init) {
+			return send_nack(res, ERROR_MESSAGES.ON_INIT_DOES_NOT_EXISTED);
+		}
+		return intializeRequest(res, next, on_init, scenario, providersItems);
+	} catch (error) {
+		return next(error);
+	}
+};
+
+const intializeRequest = async (
+	res: Response,
+	next: NextFunction,
+	transaction: any,
+	scenario: string,
+	providersItems: any
+) => {
+	try {
+		let {
+			context,
+			message: {
+				order: { provider,quote, locations, payment, fulfillments, xinput, items },
```

Comment:
Please correct any such occurences. `locations` is supposed to come inside `provider`

Reply:
Fixed in latest commit